app/services/romz.py

```
@@ -201,6 +201,27 @@ def _resolve_single_rom(
+        row-actions to archives this tool can actually handle — instead of
+        offering them on *every* ``.7z``/``.zip`` purely on extension. It
+        delegates to :meth:`_single_rom_member` so the gate applies the **same**
+        acceptance criteria as the verify/extract paths: exactly one handheld-ROM
+        payload, no symlink member, no traversal/absolute member path, and within
+        the shared archive entry/size limits. Anything those paths would reject
+        (and any unreadable/corrupt archive) is simply not romz-ready — never
+        raises, returns ``False``.
+        """
+        try:
+            cls._single_rom_member(archive_path)
+        except Exception:  # unreadable/corrupt, multi-file, unsafe, over-limit
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)